File: test/Objective-C_FuckJSON_FuckHttp/Objective-C_FuckJSON_FuckHttp/FuckJSON.m
//
//  FuckJSON.m
//  testPOSTJSONinOC
//
//  Created by caoyuan on 7/8/14.
//  Copyright (c) 2014 caoyuan. All rights reserved.
//

#import "FuckJSON.h"

@implementation FuckJSON
+(NSData *)dicToJSON:(NSMutableDictionary*)mutableDic
{
    NSData *data = [NSJSONSerialization dataWithJSONObject:mutableDic options:NSJSONWritingPrettyPrinted error:nil];
    return data;
}

+(NSMutableDictionary*)JSONtoDic:(NSData*)json
{
    NSString *responseString = [[NSString alloc] initWithData:json encoding:NSUTF8StringEncoding];
    NSData *strData = [responseString dataUsingEncoding:NSUTF8StringEncoding allowLossyConversion:YES];
    NSMutableDictionary *mutableDic = (NSMutableDictionary*)[NSJSONSerialization JSONObjectWithData:strData options:NSJSONReadingMutableLeaves error:nil];
    return mutableDic;
}

+(NSMutableArray*)JSONtoArray:(NSData*)json
{
    NSString *responseString = [[NSString alloc] initWithData:json encoding:NSUTF8StringEncoding];
    NSData *strData = [responseString dataUsingEncoding:NSUTF8StringEncoding allowLossyConversion:YES];
    NSMutableArray *mutableDic = (NSMutableArray*)[NSJSONSerialization JSONObjectWithData:strData options:NSJSONReadingMutableLeaves error:nil];
    return mutableDic;
}

@end

File: test/ShanBay/ShanBay/FuckJSON.swift
//
//  FuckJSON.swift
//  test_Swfit_post_JSON
//
//  Created by caoyuan on 7/8/14.
//  Copyright (c) 2014 caoyuan_mini. All rights reserved.
//

import UIKit

class FuckJSON: NSObject {

    class func fuckEverythingToJson(everythingYouWantFuck:AnyObject?) ->NSData?{
        var data:NSData = NSJSONSerialization.dataWithJSONObject(everythingYouWantFuck, options: NSJSONWritingOptions.PrettyPrinted, error: nil)
        return data
    }

    class func fuckJsonToDic(jsonData:NSData?)->NSDictionary {
        var responseString = NSString(data: jsonData, encoding: NSUTF8StringEncoding)
        var strData:NSData = responseString.dataUsingEncoding(NSUTF8StringEncoding, allowLossyConversion: true)
        var dic:NSDictionary = NSJSONSerialization.JSONObjectWithData(strData, options: NSJSONReadingOptions.MutableLeaves, error: nil) as NSDictionary
        return dic
    }

    class func fuckJsonToArray(jsonData:NSData?)->NSArray {
        var responseString = NSString(data: jsonData, encoding: NSUTF8StringEncoding)
        var strData:NSData = responseString.dataUsingEncoding(NSUTF8StringEncoding, allowLossyConversion: true)
        var ary:NSArray = NSJSONSerialization.JSONObjectWithData(strData, options: NSJSONReadingOptions.MutableLeaves, error: nil) as NSArray
        return ary
    }
}
